### PR TITLE
[conformance] Avoid killing by memory error

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/tests/ie_test_utils/common_test_utils/crash_handler.cpp [hunk 1]
```diff
@@ -5,11 +5,16 @@
...
+#ifndef _WIN32
+#include <sys/resource.h>
+#endif
+
 namespace CommonTestUtils {
 
 // enviroment to restore in case of crash
 jmp_buf env;
 unsigned int CrashHandler::MAX_TEST_WORK_TIME = UINT_MAX;
+unsigned long long CrashHandler::MAX_VIRTUAL_MEMORY = 0;
```
Comment (iefode) [inline src/tests/ie_test_utils/common_test_utils/crash_handler.cpp:17]: RAM size?

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/gflag_config.hpp [hunk 2]
```diff
@@ -37,6 +37,7 @@ static const char extract_body_message[] = "Optional. Allows to count extracted
 static const char shape_mode_message[] = "Optional. Allows to run `static`, `dynamic` or both scenarios. Default value is empty string allows to run both"
                                          " scenarios. Possible values are `static`, `dynamic`, ``";
 static const char test_timeout_message[] = "Optional. Setup timeout for each test in seconds, default timeout 900seconds (15 minutes).";
+static const char memory_limit_message[] = "Setup memory usage limit of virtual memory (address space) in KB, memory is unlimited by default.";
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/gflag_config.hpp:40]: `is unlimited by default` -> `is limited  by RAM size by default`

File: src/tests/ie_test_utils/common_test_utils/test_common.cpp [hunk 3]
```diff
@@ -42,19 +42,27 @@ inline size_t getVmSizeInKB() {
...
     if (file != nullptr) {
         char line[128];
 
         while (fgets(line, 128, file) != NULL) {
             if (strncmp(line, "VmSize:", 7) == 0) {
-                result = parseLine(line);
+                vmSize = parseLine(line);
+            }
+            if (strncmp(line, "VmHWM:", 6) == 0) {
+                rssMaxSize = parseLine(line);
```
Comment (sbalandi) [inline src/tests/ie_test_utils/common_test_utils/test_common.cpp:56]: need to fix , rssSize

File: src/tests/ie_test_utils/common_test_utils/test_common.cpp [hunk 4]
```diff
@@ -42,19 +42,27 @@ inline size_t getVmSizeInKB() {
...
         while (fgets(line, 128, file) != NULL) {
             if (strncmp(line, "VmSize:", 7) == 0) {
-                result = parseLine(line);
+                vmSize = parseLine(line);
+            }
+            if (strncmp(line, "VmHWM:", 6) == 0) {
+                rssSize = parseLine(line);
+            }
+            if (strncmp(line, "VmRSS:", 6) == 0) {
+                rssMaxSize = parseLine(line);
```
Comment (sbalandi) [inline src/tests/ie_test_utils/common_test_utils/test_common.cpp:59]: rssSize

Comment (sbalandi) [inline src/tests/ie_test_utils/common_test_utils/test_common.cpp:59]: rssSize

File: src/tests/ie_test_utils/common_test_utils/test_common.cpp [hunk 5]
```diff
@@ -42,19 +42,27 @@ inline size_t getVmSizeInKB() {
...
     if (file != nullptr) {
         char line[128];
 
         while (fgets(line, 128, file) != NULL) {
             if (strncmp(line, "VmSize:", 7) == 0) {
-                result = parseLine(line);
+                vmSize = parseLine(line);
+            }
+            if (strncmp(line, "VmHWM:", 6) == 0) {
+                rssSize = parseLine(line);
```
Comment (sbalandi) [inline src/tests/ie_test_utils/common_test_utils/test_common.cpp:56]: rssMaxSize

Comment (sbalandi) [inline src/tests/ie_test_utils/common_test_utils/test_common.cpp:56]: rssMaxSize

**Pull request closed without merging**